src/Chainweb/Chainweb.hs

```diff
@@ -646,43 +647,64 @@ runChainweb cw = do
 
     -- P2P Server
 
-    serverSettings :: Settings
-    serverSettings = setOnException
-        (\r e -> when (shouldDisplayException e) (logg Warn $ loggServerError r e))
-        $ peerServerSettings (_peerResPeer $ _chainwebPeer cw)
-      where
-        shouldDisplayException e
-            | Just InsecureConnectionDenied <- fromException e = False
-            | otherwise = defaultShouldDisplayException e
+    serverSettings :: IORef Int -> Settings
+    serverSettings closedConnectionsRef = setOnException
+        (\r e -> if
+            | Just InsecureConnectionDenied <- fromException e ->
+                return ()
+            | Just ClientClosedConnectionPrematurely <- fromException e ->
+                atomicModifyIORef' closedConnectionsRef (\n -> (n + 1, ()))
+            | otherwise ->
+                when (defaultShouldDisplayException e) $
+                    logg Warn $ loggServerError r e
+        ) $ peerServerSettings (_peerResPeer $ _chainwebPeer cw)
+
+    monitorConnectionsClosedByClient :: IORef Int -> IO ()
+    monitorConnectionsClosedByClient closedConnectionsRef =
+        runForever logg "ConnectionClosedByClient.counter" $ do
+            closedConnections <- atomicModifyIORef' closedConnectionsRef (\n -> (0, n))
+            when (closedConnections /= 0) $
+                logg Info $ "Clients closed connections prematurely " <> sshow closedConnections <> " times"
```

Comment:
A disadvantage of this textual encoding is that it is hard to process with elasticsearch (and similar) tools. For instance, we have a panel on the dashboard that counts connections failures of different types, which would require to write an extra ingress rule to parse the message and extract the value.

One solution would be to use the predefined counter type from `Chainweb.Counters` along with a dedicated telemetry logger. This would also allow to add more counters (e.g. for other failure types).

Though I am not sure. It might be overkill to create a dedicated telemetry logger and elasticsearch index just for a single counter value.

Reply:
How about now, using the existing data stream (telemetry logger) just with a different string attached to the counter?